pkg/reconciler/buildrun/resources/taskrun.go

```diff
@@ -234,6 +235,14 @@ func GenerateTaskRun(
 		},
 	}
 
+	// Merge Build and BuildRun NodeSelectors, giving preference to BuildRun NodeSelector
+	taskRunNodeSelector := mergeMaps(build.Spec.NodeSelector, buildRun.Spec.NodeSelector)
+	if len(taskRunNodeSelector) > 0 {
+		expectedTaskRun.Spec.PodTemplate = &pod.PodTemplate{
+			NodeSelector: taskRunNodeSelector,
+		}
+	}
```

Comment:
style nit (not blocking): we should initialize the `expectedTaskRun` pod template to an empty instance earlier than here. This will help when future contributors add the other parts of SHIP-0039.